src/core/backbone_router/bbr_leader.hpp

```diff
@@ -176,8 +189,9 @@ class Leader : public InstanceLocator
     void LogDomainPrefix(DomainPrefixState, const otIp6Prefix &) const {}
 #endif
 
-    BackboneRouterConfig mConfig;       ///< Primary Backbone Router information.
-    otIp6Prefix          mDomainPrefix; ///< Domain Prefix in the Thread network.
+    BackboneRouterConfig mConfig;               ///< Primary Backbone Router information.
+    otIp6Prefix          mDomainPrefix;         ///< Domain Prefix in the Thread network.
+    otIp6Prefix          mPreviousDomainPrefix; ///< Previous Domain Prefix that was in the Thread network.
```

Comment:
Need to cache the deprecated domain prefix somewhere, which is used to remove the deprecated DUAs registered by Child when DomainPrefix changes in
`void DuaManager::HandleDomainPrefixUpdate(BackboneRouter::Leader::DomainPrefixState aState)`

Reply:
One quick question - should the parent accept the compressed unicast address with `deprecated` prefix from the child? previous thought is the parent would not accept it, but after re-check the code, there seems no check on the compressflag of context

Reply:
Previous thought is based on the assumption that the address with `deprecated` prefix won't be accept by the parent, however it turns out to be not true in existing OT.

Reply:
Checked the spec again that 
> CID_compress is a Boolean value. If set to 'true', this context MAY be used for compressing outgoing messages. If not set, this context is only to be used for uncompressing incoming messages.

I would assume it indicates uncompressing the IP header,  does it also include uncompressing the AR TLV?

Reply:
I agree the spec is not clear on this. However, if the AR TLV address is compressed with a context that the parent does not know, I assume the parent will just fail to register that address.

Reply:
Previous design is letting the parent to remove the DUA with deprecated Domain Prefix of it child as the parent won't take stale DUA with deprecated Domain Prefix from the child as DUA
But after second thought, clearing Child DUA relative masks on the parent should be enough. Will update.